packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -301,10 +318,75 @@ abstract class FileSystemBase extends Resource implements IFileSystem {
    * @param actions The actions to grant
    */
   public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
-    return iam.Grant.addToPrincipal({
+    return iam.Grant.addToPrincipalOrResource({
+      grantee: grantee,
+      actions: actions,
+      resourceArns: [this.fileSystemArn],
+      resource: this,
+    });
+  }
+
+  /**
+   * Grant the client actions defined in actions to the given grantee on this File System resource.
+   * If this method is used and the allowAnonymousAccess props are not specified,
+   * anonymous access to this file system is prohibited.
+   *
+   * @param grantee The principal to grant right to
+   * @param actions The client actions to grant
+   * @param conditions The conditions to grant
+   * @internal
+   */
+  public _grantClient(grantee: iam.IGrantable, actions: ClientAction[], conditions?: Record<string, Record<string, unknown>>): iam.Grant {
+    this._grantedClient = true;
+    return iam.Grant.addToPrincipalOrResource({
       grantee: grantee,
       actions: actions,
       resourceArns: [this.fileSystemArn],
+      resource: this,
+      conditions,
+    });
+  }
+
+  /**
+   * Grant read permissions for this file system to an IAM principal.
+   * @param grantee The principal to grant read to
+   */
+  public grantReadBeta1(grantee: iam.IGrantable): iam.Grant {
```

Comment:
I think it is safe to remove the `Beta1` from these. Based on the API I can't
foresee any breaking changes that we would need to make.